lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -600,6 +601,13 @@ func (d *zfs) CreateVolumeFromCopy(vol Volume, srcVol Volume, copySnapshots bool
 func (d *zfs) CreateVolumeFromMigration(vol Volume, conn io.ReadWriteCloser, volTargetArgs migration.VolumeTargetArgs, preFiller *VolumeFiller, op *operations.Operation) error {
 	// Handle simple rsync and block_and_rsync through generic.
 	if volTargetArgs.MigrationType.FSType == migration.MigrationFSType_RSYNC || volTargetArgs.MigrationType.FSType == migration.MigrationFSType_BLOCK_AND_RSYNC {
+		// Round up the volume size to avoid running into "no space left on device".
+		if volTargetArgs.VolumeSize > 0 {
+			volSizeBytes := math.Ceil(float64(volTargetArgs.VolumeSize) / MinBlockBoundary * MinBlockBoundary)
```

Comment:
The ZFS `createVolume()` function already rounds anyway https://github.com/lxc/lxd/blob/master/lxd/storage/drivers/driver_zfs_utils.go#L58, so if there is a fix that is needed it should probably be in there.

This is used by `CreateVolume()` which is called by `genericVFSCreateVolumeFromMigration`